src/plugins/visualize/public/application/utils/get_top_nav_config.tsx

```diff
@@ -245,6 +256,117 @@ export const getTopNavConfig = (
+                      {
+                        defaultMessage: 'Unable to get workspaces with write permission',
+                      }
+                    ),
+                    text: error instanceof Error ? error.message : JSON.stringify(error),
+                  });
+                }
+                if (result?.success) {
+                  return result.result?.workspaces ?? [];
+                } else {
+                  return [];
+                }
+              };
+
+              const onDuplicate = async (
+                visualizationSavedObjects: SavedObjectWithMetadata[],
+                includeReferencesDeep: boolean,
+                targetWorkspace: string
+              ) => {
+                // object must be visualization type
+                const objectsToDuplicate = visualizationSavedObjects.map((obj) => ({
+                  id: obj.id,
+                  type: 'visualization',
+                }));
+
+                try {
+                  await duplicateSavedObjects(
+                    http,
+                    objectsToDuplicate,
+                    includeReferencesDeep,
+                    targetWorkspace
+                  );
+                } catch (e) {
+                  notifications.toasts.addDanger({
+                    title: i18n.translate('visualize.topNavMenu.duplicate.dangerNotification', {
+                      defaultMessage: 'Unable to duplicate visualization',
+                    }),
+                  });
+                  throw e;
+                }
+                notifications.toasts.addSuccess({
+                  title: i18n.translate('visualize.topNavMenu.duplicate.successNotification', {
+                    defaultMessage: 'Duplicate visualization successfully',
+                  }),
+                });
+              };
+
+              const visualizationSavedObject = (savedVis || {}) as SavedObjectWithMetadata;
+              visualizationSavedObject.meta = { title: savedVis?.title }; // meta is missing in savedVis
+
+              const duplicateModal = (
+                <SavedObjectsDuplicateModal
+                  duplicateState={DuplicateState.Single}
+                  selectedSavedObjects={[visualizationSavedObject]}
+                  workspaces={workspaces}
+                  getDuplicateWorkspaces={getDuplicateWorkspaces}
+                  onDuplicate={onDuplicate}
+                  onClose={() => {}}
+                />
+              );
+              const isSaveAsButton = anchorElement.classList.contains('saveAsButton');
+              onAppLeave((actions) => {
+                return actions.default();
+              });
+              if (
+                originatingApp === 'dashboards' &&
+                dashboard.dashboardFeatureFlagConfig.allowByValueEmbeddables &&
+                !isSaveAsButton
+              ) {
+                createVisReference();
```

Comment:
Q: Could you leave a comment on when and why it should call `createVisReference`?